internal/namespaces/baremetal/v1alpha1/custom_server_install.go

```diff
@@ -2,17 +2,68 @@ package baremetal
 
 import (
 	"context"
+	"reflect"
 
 	"github.com/scaleway/scaleway-cli/internal/core"
+	account "github.com/scaleway/scaleway-sdk-go/api/account/v2alpha1"
 	baremetal "github.com/scaleway/scaleway-sdk-go/api/baremetal/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
 )
 
 func serverInstallBuilder(c *core.Command) *core.Command {
 
+	type baremetalInstallServerRequestCustom struct {
+		baremetal.InstallServerRequest
+		AllSSHKeys bool
+	}
+
+	c.ArgsType = reflect.TypeOf(baremetalInstallServerRequestCustom{})
+
+	c.ArgSpecs.AddBefore("ssh-key-ids.{index}", &core.ArgSpec{
+		Name:       "all-ssh-keys",
+		Short:      "Add all SSH keys on your baremetal instance (cannot be used with ssh-key-ids)",
+		Default:    core.DefaultValueSetter("false"),
+		OneOfGroup: "ssh",
+	})
+
+	c.ArgSpecs.GetByName("ssh-key-ids.{index}").OneOfGroup = "ssh"
+
+	c.Interceptor = func(ctx context.Context, argsI interface{}, runner core.CommandRunner) (interface{}, error) {
+		tmpRequest := argsI.(*baremetalInstallServerRequestCustom)
+		request := &baremetal.InstallServerRequest{
```

Comment:
no need to recreate a new request you already have access to `tmpRequest.InstallServerRequest`